### PR TITLE
chore: Add structure to automatically reset between test runs

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: sdktests/server_side_persistence_base.go [hunk 1]
```diff
@@ -1,6 +1,17 @@
 package sdktests
 
+// NOTE: You may note that these tests do not follow the same pattern as the
+// other tests in this repository.
+//
+// Historically, tests in this repository have suffered from onion-like
+// nesting. The further you have to dig, the more you cry.
+//
+// In the time honored tradition of every other language on the planet, we are
```
Comment (kinyoklion) [inline sdktests/server_side_persistence_base.go:9]: image

File: sdktests/server_side_persistence_base.go [hunk 2]
```diff
@@ -119,58 +134,560 @@ func newServerSidePersistentTests(
...
+			pollUntilFlagValueUpdated(t, client, "flag-key", context,
+				ldvalue.String("default"), ldvalue.String("fallthrough"), ldvalue.String("default"))
+
+			// Completely reset the database so there are no valid flag definitions
+			require.NoError(t, s.persistentStore.Reset())
+
+			h.RequireEventually(t,
+				checkForUpdatedValue(t, client, "flag-key", context,
+					ldvalue.String("fallthrough"), ldvalue.String("default"), ldvalue.String("default")),
+				time.Second, time.Millisecond*20, "flag value was NOT updated after cache TTL")
```
Comment (kinyoklion) [inline sdktests/server_side_persistence_base.go]: I'm getting mixed messages between the assertion message and the test description.

File: sdktests/server_side_persistence_base.go [hunk 3]
```diff
@@ -119,58 +134,560 @@ func newServerSidePersistentTests(
...
+		})
+
+		s.runWithEmptyStore(t, "caches flag forever", func(t *ldtest.T) {
+			persistence := NewPersistence()
+			persistence.SetStore(servicedef.SDKConfigPersistentStore{
+				Type: s.persistentStore.Type(),
+				DSN:  s.persistentStore.DSN(),
+			})
+			persistence.SetCache(servicedef.SDKConfigPersistentCache{
+				Mode: servicedef.CacheModeInfinite,
```
Comment (kinyoklion) [inline sdktests/server_side_persistence_base.go:301]: Infinite mode isn't a uniform capability. I guess we can suppress the test though.

File: sdktests/server_side_persistence_base.go (hunk 2 above)
Comment (keelerm84) [reply sdktests/server_side_persistence_base.go]: Tweaked.

File: sdktests/server_side_persistence_base.go (hunk 3 above)
Comment (keelerm84) [reply sdktests/server_side_persistence_base.go:301]: That's a good point. I could make it a capability, but it does kind of feel like not supporting infinite is more of a suppression thing. If you can support the other two, seems like you should be able to support this.

File: sdktests/server_side_persistence_base.go (hunk 1 above)
Comment (keelerm84) [reply sdktests/server_side_persistence_base.go:9]: This was @cwaldren-ld 's prompting. Way to be the change you want to see. 😄

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**